packages/backend-auth/src/factory.ts

```diff
@@ -192,7 +192,7 @@ class AmplifyAuthGenerator implements ConstructContainerEntryGenerator {
 
     const ssmEnvironmentEntries =
       ssmEnvironmentEntriesGenerator.generateSsmEnvironmentEntries({
-        [`${this.defaultName}_USERPOOL_ID`]:
+        [`${this.props.name ?? this.defaultName}_USERPOOL_ID`]:
```

Comment:
not sure if it belongs in this PR or a followup, but we need to go through all of our factory ctors and make this default assignment there instead of a gajillion different places

Reply:
Okay. I think there's still an opportunity for some cleanup here, but it can go separately